Language/Java/Parser.hs

```diff
@@ -1364,17 +1408,35 @@ literal =
 -- Operators
 
 preIncDecOp, prefixOp, postfixOp :: P (Exp -> Exp)
-preIncDecOp =
-  (tok Op_PPlus >> return PreIncrement)
-    <|> (tok Op_MMinus >> return PreDecrement)
+preIncDecOp = do
+  startLoc <- getLocation
+  ( do
+      tok Op_PPlus
```

Comment:
Schreib doch mal eine Hilfsfunktion `attrTok`, die neben dem Token, den sie parset, noch das Attribut erhält, das sie zurückliefert. Dann kann man Code der folgenden Art verwenden.

```haskell
startLoc <- getLocation
constr <- (attrTok Op_PPlus PreIncrement <|> attrTok ...)
endLoc <- getLocation
return (constr (startLoc, endLoc))
```

Reply:
Soll in dem Zug auch gleich überall die neue Hilfsfunktion verwendet werden?
Sehr viele Stellen im Code verwenden ja genau das Muster, was jetzt die Hilfsfunktion implementiert:
```
tok token >> return contructor
```

Reply:
Ah, nee, gerade nach dem Schreiben fiel mir auf, dass das nicht wirklich was bringt.